tools/shared.py

```diff
@@ -1258,46 +1255,6 @@ def consider_object(f, force_add=False):
         actual_files.append(f)
       return do_add
 
-    def get_archive_contents(f):
-      if f in ar_contents:
-        return ar_contents[f]
-
-      cwd = os.getcwd()
-      try:
-        emscripten_temp_dir = get_emscripten_temp_dir()
-        temp_dir = os.path.join(emscripten_temp_dir, 'ar_output_' + str(os.getpid()) + '_' + str(len(temp_dirs)))
-        temp_dirs.append(temp_dir)
-        safe_ensure_dirs(temp_dir)
-        os.chdir(temp_dir)
-        contents = filter(lambda x: len(x) > 0, Popen([LLVM_AR, 't', f], stdout=PIPE).communicate()[0].split('\n'))
-        # llvm-ar appears to just use basenames inside archives. as a result, files with the same basename
-        # will trample each other when we extract them. to help warn of such situations, we warn if there
-        # are duplicate entries in the archive
-        if len(contents) != len(set(contents)):
-          logging.warning('loading from archive %s, which has duplicate entries (files with identical base names). this is dangerous as only the last will be taken into account, and you may see surprising undefined symbols later. you should rename source files to avoid this problem (or avoid .a archives, and just link bitcode together to form libraries for later linking)' % f)
```

Comment:
It's not gone away, it's just been split out into a separate function further down the file (`ar_contents`). I didn't even make any changes to this, I just needed it to be an independent function so I could call it elsewhere.

(Actually, I did change it by mistake when rebasing, I lost a few lines of changes that had been made up here. I've put those back now.)